CMakeLists.txt

```diff
@@ -101,12 +103,17 @@ install(FILES snappy.h
         ${Snappy_BINARY_DIR}/snappy-stubs-public.h
         DESTINATION include)
 
+if (BUILD_SHARED_LIBS)
```

Comment:
There is no `BUILD_STATIC_LIBS`, `BUILD_SHARED_LIBS` [is a CMake variable](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html). 
You would have to add 2 libraries and create targets with different names for each
combined with custom logic with enables or disables it. 
downstream projects can’t use both at the same time anyway
The way it is now downstream gets shared snappy by default and would get static snappy when BUILD_SHARED_LIBS is set (which i consider the build-all-my-stuff-statically-please flag) which has the benefit that downstream doesn’t have to explicitly state anything in their CMake file, it’s only a build option. 
One could also add explicit aliases on top of that (in the future?). 
Reconfiguring in between has no downside anyway, it’s just a stylistic issue. 
I’m quite happy with the way it is now, I also don’t know of any CMake projects which handle building static and shared flavors nicely to compare it with.